### PR TITLE
Remove outdated Mercurial sections

### LINKED ISSUE #796: Drop Mercurial sections
Suggesting to drop the following two sections:

1. [Git Bootcamp, Applying a Patch from Mercurial to Git](https://devguide.python.org/gitbootcamp/#applying-a-patch-from-mercurial-to-git)
2. [Triaging, Mercurial Repository](https://devguide.python.org/triaging/#mercurial-repository)

Maybe instead of dropping the Mercurial Repository section from the Triaging chapter, we could just note that this BPO field is not used anymore.

### PR DESCRIPTION
**Pull request by erlend-aasland**
Keep _Mercurial Repository_ in _Triaging_, since BPO links to it.

Resolves #796

### CODE REVIEW

File: triaging.rst [hunk 1]
```diff
@@ -505,13 +505,7 @@ with the "open" status.
...
 Mercurial Repository
 ''''''''''''''''''''
-HTTP link to a Mercurial repository that contains a patch for the issue.
-A :guilabel:`Create Patch` button will appear that computes a diff for the
-head revision of the remote branch and attaches it to the issue.  The button
-supports only CPython_ patches.
-
-If you don't indicate a remote branch, ``default`` is used.  You can
-indicate a remote branch by adding ``#BRANCH`` to the end of the URL.
+This field is not used anymore.
```
Comment (zware) [inline triaging.rst:508]:
```suggestion
Deprecated: HTTP link to a Mercurial repository that contains a patch for the issue.

New repository links should not be added to new or existing issues.
```

I also just opened a PR on the tracker to remove the ability to add new hg repo links: psf/bpo-tracker-cpython#8.  If that gets merged (and works :)), we could just remove this section.

**Review by zware — CHANGES_REQUESTED** [review]

File: triaging.rst (hunk 1 above)
Comment (erlend-aasland) [reply triaging.rst:508]: Great!

File: gitbootcamp.rst [hunk 2]
```diff
@@ -318,46 +318,6 @@ When it happens, you need to resolve conflict.  See these articles about resolvi
...
 - `Resolving a merge conflict using the command line <https://help.github.com/en/articles/resolving-a-merge-conflict-using-the-command-line>`_
 
-.. _git_from_mercurial:
-
-Applying a Patch from Mercurial to Git
---------------------------------------
-
-Scenario:
-
-- A Mercurial patch exists but there is no pull request for it.
```
[the remark this answers is not in the document]
Comment (erlend-aasland) [reply gitbootcamp.rst:328]: Well, I guess that still happens, so it makes sense to keep it. I'll reword it slightly.

File: pullrequest.rst [hunk 3]
```diff
@@ -394,8 +394,6 @@ Author Name <email_address> ." to the pull request description and commit messag
 See `the GitHub article <https://help.github.com/articles/creating-a-commit-with-multiple-authors/>`_
 on how to properly add the co-author info.
 
-See also :ref:`Applying a Patch from Mercurial to Git <git_from_mercurial>`.
```
Comment (zware) [inline pullrequest.rst:397]: We should keep this pointer, with corrected title and anchor.

**Review by zware — CHANGES_REQUESTED** [review]

**Review by zware — APPROVED** [review]

**Comment by erlend-aasland** [comment]
Thanks for improving this PR, Zach.

**Pull request merged**